packages/react-router-dom/dom.ts

```diff
@@ -176,34 +177,21 @@ export function getFormSubmissionInfo(
   payload: any;
 } {
   let method: string;
-  let action: string | null = null;
+  let action: string | null;
   let encType: string | null;
   let formData: FormData | undefined = undefined;
   let payload: unknown = undefined;
 
   if (isFormElement(target)) {
-    let submissionTrigger: HTMLButtonElement | HTMLInputElement = (
-      options as any
-    ).submissionTrigger;
-
-    if (options.action) {
-      action = options.action;
-    } else {
```

Comment:
For all cases (`action`/`encType`/`method`/etc.), `options[field]` took precedence, and now that `action` could be an `ActionFunction` it made sense to pull that out - so all `options[field]` overrides happen in the body of useSubmit now instead of in `getFormSubmissionInfo`.  This also better prepares us for pulling `useSubmit` down into `react-router` where it can skip the DOM stuff.